tox.ini

```
@@ -94,7 +94,7 @@ commands_post =
       browse_cmd = shlex.join(("python3", "-Im", "webbrowser", html_url)); \
       serve_cmd = shlex.join((\
         "python3", "-Im", "http.server", \
-        "--directory", "cov_html_report_dir", "0", \
+        "--directory", f"{cov_html_report_dir}", "0", \
```

`cov_html_report_dir` does not work, so I fall back to `f"{cov_html_report_dir}"`

```bash
py: commands_post[2]> .tox/py/bin/python -bb -E -s -I -Werror '-cimport pathlib, shlex, sys; cov_html_report_arg_prefix = "--cov-report=html:"; cov_html_reports = [arg[len(cov_html_report_arg_prefix):] for arg in sys.argv if arg.startswith(cov_html_report_arg_prefix)]; cov_html_reports or sys.exit(); cov_html_report_dir = pathlib.Path(cov_html_reports[-1]); index_file = cov_html_report_dir / "index.html";html_url = f"file://{index_file}";browse_cmd = shlex.join(("python3", "-Im", "webbrowser", html_url)); serve_cmd = shlex.join(("python3", "-Im", "http.server", "--directory", cov_html_report_dir, "0", )); print(f"\nTo open the HTML coverage report, run\n\n\t{browse_cmd !s}\n");print(f"To serve the HTML coverage report with a local web server, use\n\n\t{serve_cmd !s}\n")' --cov-report=html:/home/vm/code/0-other/open-source/pre-commit-terraform/.tox/py/tmp/htmlcov/
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.10/shlex.py", line 320, in join
    return ' '.join(quote(arg) for arg in split_command)
  File "/usr/lib/python3.10/shlex.py", line 320, in <genexpr>
    return ' '.join(quote(arg) for arg in split_command)
  File "/usr/lib/python3.10/shlex.py", line 329, in quote
    if _find_unsafe(s) is None:
TypeError: expected string or bytes-like object
py: exit 1 (0.03 seconds) /home/vm/code/0-other/open-source/pre-commit-terraform> .tox/py/bin/python -bb -E -s -I -Werror '-cimport pathlib, shlex, sys; cov_html_report_arg_prefix = "--cov-report=html:"; cov_html_reports = [arg[len(cov_html_report_arg_prefix):] for arg in sys.argv if arg.startswith(cov_html_report_arg_prefix)]; cov_html_reports or sys.exit(); cov_html_report_dir = pathlib.Path(cov_html_reports[-1]); index_file = cov_html_report_dir / "index.html";html_url = f"file://{index_file}";browse_cmd = shlex.join(("python3", "-Im", "webbrowser", html_url)); serve_cmd = shlex.join(("python3", "-Im", "http.server", "--directory", cov_html_report_dir, "0", )); print(f"\nTo open the HTML coverage report, run\n\n\t{browse_cmd !s}\n");print(f"To serve the HTML coverage report with a local web server, use\n\n\t{serve_cmd !s}\n")' --cov-report=html:/home/vm/code/0-other/open-source/pre-commit-terraform/.tox/py/tmp/htmlcov/ pid=613518
.pkg: _exit> python /home/vm/snap/code/179/.local/share/uv/tools/tox/lib/python3.10/site-packages/pyproject_api/_backend.py True hatchling.build
  py: FAIL code 1 (0.91=setup[0.37]+cmd[0.47,0.01,0.03,0.03] seconds)
  evaluation failed :( (0.97 seconds)
  ```

---

> `cov_html_report_dir` does not work, so I fall back to `f"{cov_html_report_dir}"`

Doesn't work probably because it is wrapped in quotes and hence is string rather than var value? 🤔

---

Just in case I did get the context correct:
```suggestion
        "--directory", cov_html_report_dir, "0", \
```

---

`shlex.join()` doesn't like the raw var because it's a pathlib object. Maybe, it'd be fine on more modern Pythons. But regardless, you can just make it a string:
```suggestion
        "--directory", str(cov_html_report_dir), "0", \
```

---

Ah, right, explicit conversion to string is what I missed in my suggestion and what worked for Max as f-string coz by default f-string interprets value as string I guess.
@webknjaz Thanks for looking after and helping remediate my silly errors ⭐

---

Well, it was me who made this mistake initially, when I was stripping off parts of my usual setup :)